src/ServiceControl.Audit/Infrastructure/WebApi/WebApiHostBuilderExtensions.cs

```diff
@@ -41,20 +41,8 @@ static void RegisterInternalWebApiControllers(ContainerBuilder containerBuilder)
 
             foreach (var controllerType in controllerTypes)
             {
-                containerBuilder.RegisterType(controllerType).FindConstructorsWith(new AllConstructorFinder());
+                containerBuilder.RegisterType(controllerType);
```

Comment:
I wonder if this is needed anymore. Will the ASP.NET infra already add these controllers now? I'm guessing that's the next step.

Reply:
Not sure if this is what you are referring to @mikeminutillo: according to this [documentation](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.applicationparts.applicationpartattribute?view=aspnetcore-6.0) as long as a library references MVC, an instance of ApplicationPartAttribute will be auto generated and the library will be treated as an application part

Reply:
I'll check if we can just remove the explicit registration

Reply:
If I remove these registrations, the web server throws 500 errors because of missing empty actors. I'm not sure whether this is a limitation of older versions of web API, OWIN, Autofac or something else. I'll add a note to our POA to see if would still be needed without Autofac.

Reply:
No worries. I just figured that most ASP.NET applications don't have to manually register controllers individually so we must be doing something funky. Maybe we need an `.AddControllers()` somewhere or something.